packages/pubsub/src/index.js

```diff
@@ -79,43 +67,156 @@ function PubSub(options) {
+ *     message that you must ack a message before it is redelivered.
+ * @param {object} options.flowControl - Flow control configurations for
+ *     receiving messages.
+ * @param {number} options.flowControl.maxBytes - The maximum number of bytes
+ *     in un-acked messages to allow before the subscription pauses incoming
+ *     messages. Defaults to 20% of free memory.
+ * @param {number} options.flowControl.maxMessages - The maximum number of
+ *     un-acked messages to allow before the subscription pauses incoming
+ *     messages. Default: Infinity.
+ * @param {object} options.gaxOpts - Request configuration options, outlined
+ *     here: https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {number|date} options.messageRetentionDuration - Set this to override
+ *     the default duration of 7 days. This value is expected in seconds.
+ *     Acceptable values are in the range of 10 minutes and 7 days.
+ * @param {string} options.pushEndpoint - A URL to a custom endpoint that
+ *     messages should be pushed to.
+ * @param {boolean} options.retainAckedMessages - If set, acked messages are
+ *     retained in the subscription's backlog for 7 days (unless overriden by
+ *     `options.messageRetentionDuration`). Default: `false`
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request
+ * @param {module:pubsub/subscription} callback.subscription - The subscription.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * //-
+ * // Subscribe to a topic. (Also see {module:pubsub/topic#createSubscription}).
+ * //-
+ * var topic = 'messageCenter';
+ * var name = 'newMessages';
+ *
+ * var callback = function(err, subscription, apiResponse) {};
+ *
+ * pubsub.createSubscription(topic, name, callback);
+ *
+ * //-
+ * // Customize the subscription.
+ * //-
+ * pubsub.createSubscription(topic, name, {
+ *   ackDeadline: 90000 // 90 seconds
+ * }, callback);
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * pubsub.createSubscription(topic, name).then(function(data) {
+ *   var subscription = data[0];
+ *   var apiResponse = data[1];
+ * });
+ */
+PubSub.prototype.createSubscription = function(topic, name, options, callback) {
+  if (!is.string(topic) && !(topic instanceof Topic)) {
+    throw new Error('A Topic is required for a new subscription.');
+  }
+
+  if (!is.string(name)) {
+    throw new Error('A subscription name is required.');
+  }
+
+  if (is.string(topic)) {
+    topic = this.topic(topic);
+  }
+
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  options = options || {};
+
+  var subscription = this.subscription(name, options);
+
+  var reqOpts = extend({
+    topic: topic.name,
+    name: subscription.name
+  }, options);
+
+  delete reqOpts.gaxOpts;
+
+  if (options.ackDeadline) {
+    reqOpts.ackDeadlineSeconds = options.ackDeadline / 1000;
```

Comment:
@callmehiphop Can you point me to the code where the p99 value is used to update this value (if the user does not specify something)?